### PR TITLE
New Algorithm: Pruned Landmark Labeling

### PR DESCRIPTION
**Pull request by angriman**
This PR brings the Pruned Landmark Labeling algorithm by Akiba et al. [1]. The algorithm computes distance labels which are used to answer shortest-path distance queries.

I wrote this code a few years ago and, finally, I found some time to polish it.

[1] https://dl.acm.org/doi/10.1145/2463676.2465315

### CODE REVIEW

File: include/networkit/distance/PrunedLandmarkLabeling.hpp [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+class PrunedLandmarkLabeling : public Algorithm {
+
+public:
+    /**
+     * Pruned Landmark Labeling algorithm based on the paper "Fast exact shortest-path distance
+     * queries on large networks by pruned landmark labeling" from Akiba et al., ACM SIGMOD 2013.
+     * The algorithm computes distance labels by performing pruned breadth-first searches from each
+     * vertex. Labels are used to quickly retrieve shortest-path distances between node pairs.
+     * Note: this algorithm only works for unweighted graphs.
```
Comment (fabratu) [inline include/networkit/distance/PrunedLandmarkLabeling.hpp:20]: Better use @note.

File: networkit/distance.pyx [hunk 2]
```diff
@@ -1791,3 +1791,53 @@ cdef class ReverseBFS(SSSP):
...
+		self._this = new _PrunedLandmarkLabeling(G._this)
+
+	def __dealloc__(self):
+		self._G = None
+
+	def query(self, node u, node v):
+		"""
+		query(u, v)
+
+		Returns the shortest-path distance between the two nodes.
```
Comment (fabratu) [inline networkit/distance.pyx:1829]: I think we should add some kind of feedback (or test), if `query` is called before `run`.

**Review by fabratu — CHANGES_REQUESTED** [review]
LGTM, only some minor comments. I merged the PLP-revert, you can also rebase to fix CI errors.

File: networkit/distance.pyx (hunk 2 above)
Comment (angriman) [reply networkit/distance.pyx:1829]: I added `assureFinished()` in the `.cpp` file.

**Pull request merged**